authentik/sources/ldap/sync/membership.py

```diff
@@ -47,9 +49,24 @@ def sync(self, page_data: list) -> int:
             return -1
         membership_count = 0
         for group in page_data:
-            if "attributes" not in group:
-                continue
-            members = group.get("attributes", {}).get(self._source.group_membership_field, [])
+            if self._source.lookup_groups_from_user:
+                group_dn = group.get("dn", {})
+                group_filter = f"({self._source.group_membership_field}={group_dn})"
```

Comment:
Do we need to consider re-using the user filter here as well?

Reply:
Actually probably not needed since we filter users afterwards anyway. But might incur a performance impact for big directories only syncing a handful of users